migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs

```diff
@@ -94,6 +94,18 @@ impl PostgresFlavour {
 #[async_trait::async_trait]
 impl SqlFlavour for PostgresFlavour {
     async fn acquire_lock(&self, connection: &Connection) -> ConnectorResult<()> {
+        let is_cockroach = connection
+            .version()
+            .await?
+            .map(|v| v.contains("CockroachDB"))
+            .unwrap_or(false);
```

Comment:
We're going to have to split the responsibilities between connector, flavour and connection better, soon. We _should_ have the circumstances here. But it's fine for now.

Reply:
Also, maybe we could just put the link from the PR description into a comment here

Reply:
Pushed a better version.